ext/datasource/nacos/nacos.go

```diff
@@ -0,0 +1,101 @@
+package nacos
+
+import (
+	"fmt"
+
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/nacos-group/nacos-sdk-go/clients/config_client"
+	"github.com/nacos-group/nacos-sdk-go/vo"
+	"github.com/pkg/errors"
+)
+
+var (
+	logger = logging.GetDefaultLogger()
+)
+
+type NacosDataSource struct {
+	datasource.Base
+	client         config_client.IConfigClient
+	isInitialized  util.AtomicBool
+	propertyConfig vo.ConfigParam
+	closeChan      chan struct{}
+}
+
+func NewNacosDataSource(client config_client.IConfigClient, propertyConfig vo.ConfigParam, handlers ...datasource.PropertyHandler) (*NacosDataSource, error) {
```

Comment:
Maybe we could just replace `vo.ConfigParam` with `dataId` and `groupId` here? Using `vo.ConfigParam` might be confusing (e.g. the listener function inside the config is not actually used).